src/main/scala/io/iohk/ethereum/mpt/MerklePatriciaTrie.scala

```diff
@@ -118,32 +114,32 @@ class MerklePatriciaTrie[K, V] private (private val rootHash: Option[Array[Byte]
   /**
     * Implicits
     */
-  private[mpt] implicit val nodeDec = new RLPDecoder[Node] {
-    override def decode(rlp: RLPEncodeable): Node = rlp match {
+  private[mpt] implicit val nodeDec = new RLPDecoder[MptNode] {
```

Comment:
As there is no more custom `hashFn` I think we can move this away from `class MerklePatriciaTrie`